presto-main/src/main/java/io/prestosql/operator/NestedLoopJoinOperator.java

```diff
@@ -42,14 +43,23 @@
         private final int operatorId;
         private final PlanNodeId planNodeId;
         private final JoinBridgeManager<NestedLoopJoinBridge> joinBridgeManager;
+        private final List<Integer> probeChannels;
```

Comment:
We don't need explicit pruning in cross join as for cross joins pruning can be pushed down to both sides always. Cross join doesn't need any symbol for itself, but only passes all of them through.

Is there a larger benefit of this change (it adds complexity to otherwise basic operator)

Reply:
The benefit is that during optimisation we will be able to freely change JoinNode's parameters (outputSymbols, join type, filter..) without risking that the resulting JoinNode will turn out to be a cross join and fail in constructor because of output symbols pruning.

Reply:
> The benefit is that during optimisation we will be able to freely change JoinNode's parameters (outputSymbols, join type, filter..) without risking that the resulting JoinNode will turn out to be a cross join and fail in constructor because of output symbols pruning.

Non cross-join also require left symbols to be before right symbols. Optimizers can freely change output symbols order in join source nodes.

Reply:
Yes, and it is in scope of this PR to remove this requirement (left before right).
It will simplify the way that the `outputSymbols` list is prepared for `JoinNode` constructor.
One example is `PushProjectionThroughJoin.pushProjectionThroughJoin()`, another is `JoinNode.flipChildren()`.